### PR TITLE
[feature](merge-cloud) Add `BrokerLoad` and `DriverUrl` white list

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
* Add a whilte list mechanism for BrokerLoad endpoint and JDBC DriverUrl for security check
  in public cloud env

### CODE REVIEW

**Review by morningman — COMMENTED** [review]
I will review this later

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2685,6 +2685,16 @@ public static boolean isNotCloudMode() {
 
     @ConfField(mutable = true, masterOnly = false)
     public static String security_checker_class_name = "";
+
+    // the white list for the s3 load endpoint, if it is empty, no white list will be set
+    // for example: s3_load_endpoint_white_list=a,b,c
+    @ConfField(mutable = true)
+    public static String[] s3_load_endpoint_white_list = {};
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2692]: Use same name as `jdbc_driver_secure_path`.
And does this config need to be mutable?

Comment (gavinchou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2692]: It needs to be compatible with the existing warehouse, this should be discussed

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java [hunk 2]
```diff
@@ -92,6 +94,15 @@ public class LoadStmt extends DdlStmt {
     public static final String BOS_ACCESSKEY = "bos_accesskey";
     public static final String BOS_SECRET_ACCESSKEY = "bos_secret_accesskey";
 
+    // for S3 load check
+    public static final String AWS_ENDPOINT = "AWS_ENDPOINT";
```
[the remark this answers is not in the document]
Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java]: > This has been defined in S3Properties.java

resolved

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java [hunk 3]
```diff
@@ -291,22 +292,31 @@ public static String getFullDriverUrl(String driverUrl) throws IllegalArgumentEx
             String schema = uri.getScheme();
             if (schema == null && !driverUrl.startsWith("/")) {
                 return "file://" + Config.jdbc_drivers_dir + "/" + driverUrl;
```
[the remark this answers is not in the document]
Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java:301]: > local files should be check too

resolved

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Pull request merged**